pages/post.js

```diff
@@ -0,0 +1,51 @@
+import { Component } from 'react'
+import { Query } from 'react-apollo'
+import gql from 'graphql-tag'
+
+import Layout from '../components/Layout'
+import Article from '../components/Article'
+import Comment from '../components/Comment'
+import withData from '../lib/withData'
+
+class Post extends Component {
+  constructor (props) {
+    super(props)
+
+    this.articleQuery = gql`
+      query article($id: ID!) {
+        article(id:$id) {
+          id
```

Comment:
You''ll want to bring the slug in too, and use the slug to find the article instead of the ID